std/build.zig

```diff
@@ -1196,9 +1203,125 @@ pub const Target = union(enum) {
         };
     }
 
+    pub fn isNetBSD(self: Target) bool {
+        return switch (self.getOs()) {
+            .netbsd => true,
+            else => false,
+        };
+    }
+
     pub fn wantSharedLibSymLinks(self: Target) bool {
         return !self.isWindows();
     }
+
+    pub fn osRequiresLibC(self: Target) bool {
+        return self.isDarwin() or self.isFreeBSD() or self.isNetBSD();
+    }
+
+    pub fn getArchPtrBitWidth(self: Target) u32 {
```

Comment:
This is right and wrong at the same time, the effective pointer size is given by the architecture **and** the ABI: `x32` means the full 64bit registers are used but the pointer size if trimmed to 4 bytes instead of 8.